tests/reporter/kubernetes.go

```diff
@@ -918,3 +1024,157 @@ func writeStringToFile(filePath string, data string) error {
 	_, err = f.WriteString(data)
 	return err
 }
+
+func getVmiType(vmi v12.VirtualMachineInstance) (string, error) {
+	for _, volume := range vmi.Spec.Volumes {
+		if volume.VolumeSource.ContainerDisk == nil {
+			continue
+		}
+
+		image := volume.VolumeSource.ContainerDisk.Image
+		if strings.Contains(image, "fedora") {
+			return "fedora", nil
+		} else if strings.Contains(image, "cirros") {
+			return "cirros", nil
+		} else if strings.Contains(image, "alpine") {
+			return "alpine", nil
+		}
+	}
+
+	return "", fmt.Errorf("unknown type, vmi %s", vmi.ObjectMeta.Name)
+}
+
+func prepareVmiConsole(vmi v12.VirtualMachineInstance, vmiType string) error {
+	err := console.SafeExpectBatch(&vmi, []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: console.PromptExpression},
+	}, 5)
+
+	if err != nil {
+		switch vmiType {
+		case "fedora":
+			return console.LoginToFedora(&vmi)
+		case "cirros":
+			return console.LoginToCirros(&vmi)
+		case "alpine":
+			return console.LoginToAlpine(&vmi)
+		}
+	}
+
+	return err
+}
+
+func checkPodComputeContainer(pod v1.Pod) bool {
```

Comment:
nit: maybe change to `isPodComputeContainer`

Reply:
but the pod is not a compute container, it might have a compute container, this is what the function checks

Reply:
exactly, boolean functions like this one, are a lot of times called `isSomething`  or `areSomething`.
for exapmle check out the [first result](https://levelup.gitconnected.com/knot-of-nots-avoiding-negative-names-for-boolean-methods-641896a94a42#:~:text=The%20usual%20convention%20to%20name,exists%2C%20you%20would%20say%20user.) in google for "boolean function naming convention".

there is no known convention for `check Something` AFAIK, and it could mean many things.